src/lib/storage/constraints/table_unique_constraint.hpp

```diff
@@ -0,0 +1,26 @@
+#pragma once
+
+#include "table_key_constraint.hpp"
+
+namespace opossum {
+
+/**
+ * Container to define a unique constraint for a given set of column ids.
+ * Validity checks take place when unique constraints are added to tables. For example, checks for nullability in
+ * case of PRIMARY_KEY constraints.
+ */
+class TableUniqueConstraint final : public TableKeyConstraint {
```

Comment:
What is the purpose of this class? It doesn't seem to add anything to TableKeyConstraint.

Reply:
see https://github.com/hyrise/hyrise/pull/2147#issuecomment-637642816